### PR TITLE
Replace BaseVector::setSize with resize

### PR DESCRIPTION
**Pull request by kgpai**
BaseVector::setSize method used to change length_ without resizing nulls buffer
or any other buffers. Hence, calling this method might have put vector in an
inconsistent state. This change removes setSize method and updates call sites
to use resize.

### CODE REVIEW

**Comment by mbasmanova** [comment]
@kgpai Krishna, thank you for improving Vector APIs. There are CircleCI failures. Would you take a look?

File: velox/vector/DictionaryVector.h [hunk 1]
```diff
@@ -214,6 +215,18 @@ class DictionaryVector : public SimpleVector<T> {
     setInternalState();
   }
 
+  void resize(vector_size_t size, bool setNotNull = true) override {
+    if (indices_ && indices_->refCount() > 1) {
+      VELOX_FAIL("Cannot resize shared indices.");
```
Comment (mbasmanova) [inline velox/vector/DictionaryVector.h:220]: Instead of failing, can we just allocate a new buffer and copy the data? Also, can we do that only if size > current size?

File: velox/vector/tests/VectorTest.cpp [hunk 2]
```diff
@@ -1684,6 +1684,52 @@ TEST_F(VectorTest, multipleDictionariesOverLazy) {
   }
 }
 
+TEST_F(VectorTest, dictionaryResize) {
+  // Resize a simple dictionary.
+  auto vectorMaker = std::make_unique<test::VectorMaker>(pool_.get());
+  vector_size_t size = 10;
+  std::vector<int64_t> elems{1, 2, 3, 4, 5};
```
Comment (mbasmanova) [inline velox/vector/tests/VectorTest.cpp:1691]: naming: do not abbreviate: elems -> elements

File: velox/vector/DictionaryVector.h [hunk 3]
```diff
@@ -214,6 +215,18 @@ class DictionaryVector : public SimpleVector<T> {
     setInternalState();
   }
 
+  void resize(vector_size_t size, bool setNotNull = true) override {
```
Comment (mbasmanova) [inline velox/vector/DictionaryVector.h:218]: Would you document this method to explain what values are used for the "added" elements?

File: velox/vector/tests/VectorTest.cpp [hunk 4]
```diff
@@ -1684,6 +1684,52 @@ TEST_F(VectorTest, multipleDictionariesOverLazy) {
   }
 }
 
+TEST_F(VectorTest, dictionaryResize) {
```
Comment (mbasmanova) [inline velox/vector/tests/VectorTest.cpp:1687]: This test is a bit hard to follow. Would you clarify what it does?

File: velox/expression/Expr.cpp [hunk 5]
```diff
@@ -662,8 +662,7 @@ void Expr::addNulls(
     // Not all Vectors support resize.  Since we only want to append nulls,
     // we can work around that by calling setSize to resize the vector and
     // ensureNullsCapacity to resize the nulls_ bit vector.
-    result->setSize(rows.end());
-    result->ensureNullsCapacity(rows.end(), true);
+    result->resize(rows.end(), true);
```
Comment (mbasmanova) [inline velox/expression/Expr.cpp:665]: Nice. ensureNullsCapacity is not used anymore outside of the BaseVector, right? Let's make it private or fold it into the caller.

To simplify this PR, consider extracting this change + adding DictionaryVector::resize + test into a separate PR.

File: velox/vector/tests/VectorTest.cpp (hunk 4 above)
Comment (kgpai) [reply velox/vector/tests/VectorTest.cpp:1687]: I have modified this and added a few more tests.

**Review by mbasmanova — APPROVED** [review]
@kgpai Looks great. I update PR title and summary for clarity. Also, formatted summary to wrap at 80 characters. Please, re-import before landing.
